tfx/orchestration/experimental/kubernetes/kubernetes_dag_runner.py

```diff
@@ -0,0 +1,378 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Definition of kubernetes TFX runner."""
+
+import absl
+import datetime
+import re
+import time
+from typing import Optional, Text, Type
+
+from ml_metadata.proto import metadata_store_pb2
+from tfx.dsl.component.experimental import container_component
+from tfx.components.base import base_node
+from tfx.orchestration import data_types
+from tfx.orchestration import metadata
+from tfx.orchestration import pipeline as tfx_pipeline
+from tfx.orchestration import tfx_runner
+from tfx.orchestration.config import base_component_config
+from tfx.orchestration.config import config_utils
+from tfx.orchestration.config import pipeline_config
+from tfx.orchestration.kubeflow import node_wrapper
+from tfx.orchestration.kubeflow import utils
+from tfx.orchestration.launcher import base_component_launcher
+from tfx.orchestration.launcher import in_process_component_launcher
+from tfx.orchestration.launcher import kubernetes_component_launcher
+from tfx.utils import json_utils, kube_utils
+from google.protobuf import json_format
+from kubernetes import client
+import json
+
+_CONTAINER_COMMAND = [
+    'python', '/tfx-src/tfx/orchestration/experimental/kubernetes/container_entrypoint.py'
+]
+
+_DRIVER_COMMAND = [
+    'python', '/tfx-src/tfx/orchestration/experimental/kubernetes/driver_container_entrypoint.py'
+]
+
+_WRAPPER_SUFFIX = 'Wrapper'
+
+_TFX_IMAGE = 'gcr.io/tfx-eric/tfx-dev'
+
+
+def _sanitize_pod_name(pod_name: Text) -> Text:
```

Comment:
Is this copied from another location?

Reply:
Yes, it is copied from kubernetes component launcher. For consistency, I refactored both of them into kube_utils.